File: dynamic_programming/fibonacci.go
package main

import "fmt"

type Fibonacci struct {
	sequence []int
}

func NewFibonacci() *Fibonacci {
	return &Fibonacci{
		sequence: []int{0, 1},
	}
}

func (f *Fibonacci) Get(index int) []int {
	if index <= 0 {
		return []int{}
	}
	
	for len(f.sequence) < index {
		nextVal := f.sequence[len(f.sequence)-1] + f.sequence[len(f.sequence)-2]
		f.sequence = append(f.sequence, nextVal)
	}
	
	result := make([]int, index)
	copy(result, f.sequence[:index])
	return result
}

func main() {
	fib := NewFibonacci()
	fmt.Println(fib.Get(10))
}

File: dynamic_programming/fibonacci.rs
pub struct Fibonacci {
    sequence: Vec<i32>,
}

impl Fibonacci {
    pub fn new() -> Self {
        Fibonacci {
            sequence: vec![0, 1],
        }
    }

    pub fn get(&mut self, index: usize) -> Vec<i32> {
        if index == 0 {
            return Vec::new();
        }
        
        while self.sequence.len() < index {
            let next_val = self.sequence[self.sequence.len() - 1] + self.sequence[self.sequence.len() - 2];
            self.sequence.push(next_val);
        }
        
        self.sequence[..index].to_vec()
    }
}

fn main() {
    let mut fib = Fibonacci::new();
    println!("{:?}", fib.get(10));
}
